zstacklib/zstacklib/utils/ssh_validation.py

```
@@ -0,0 +1,154 @@
+                len(username), _MAX_SSH_USERNAME_LENGTH)
+        )
+    if not SSH_USERNAME_PATTERN.match(username):
+        raise SSHValidationError(
+            "SSH username format is invalid (only letters, digits, underscores(_), dots(.), hyphens(-) and trailing dollar($) allowed), current value: {0}".format(
+                username)
+        )
+
+
+def validate_ssh_port(port):
+    if port is None:
+        raise SSHValidationError("SSH port number cannot be empty")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)